### PR TITLE
Rename dump command from --dumps-dir to --dump-dir

### LINKED ISSUE #3132: CLI stabilization: rename the dumps commands
Following this discussion: https://github.com/meilisearch/product/discussions/560
Specification: WIP

As part of the [CLI stabilization](https://github.com/meilisearch/meilisearch/issues/2752), we want to rename the dump commands.

- `dumps-dir` in the config file should be renamed to `dump-dir`
- `--dumps-dir` option when launching Meilisearch binary should be renamed `--dump-dir`
- `MEILI_DUMPS_DIR` env variable should be renamed to `MEILI_DUMP_DIR`

⚠️ Don't forget to change the [default config](https://github.com/meilisearch/meilisearch/blob/main/config.toml) file as well.

## Impacted teams

@meilisearch/integration-team, @meilisearch/cloud-team and @meilisearch/docs-team 

## TODO

- [ ] Implement the changes and merge to the `main` branch
- [X] Update the related spec -> https://github.com/meilisearch/specifications/pull/204

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue
Fixes #3132 

## What does this PR do?
- Rename the dump commands, env variables and default config

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch-http/tests/common/server.rs [hunk 1]
```diff
@@ -184,7 +184,7 @@ impl Server {
 pub fn default_settings(dir: impl AsRef<Path>) -> Opt {
     Opt {
         db_path: dir.as_ref().join("db"),
-        dumps_dir: dir.as_ref().join("dump"),
+        dump_dir: dir.as_ref().join("dump"),
```
Comment (irevoire) [inline meilisearch-http/tests/common/server.rs:187]:
```suggestion
        dump_dir: dir.as_ref().join("dumps"),
```

I know this has nothing to do with your PR, but I just noticed a small inconsistency between the tests and the prod. The directory should be called `dumps` here :+1:

**Review by irevoire — DISMISSED** [review]
Hey, it looks pretty good to me!

**Review by irevoire — APPROVED** [review]
Nice, thank you! :game_die: 

bors merge

**Comment by dureuill** [comment]
Looks like MacOS tests failed spuriously, let's try again!

bors merge

**Pull request merged**